modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -371,38 +367,97 @@ def _manage_skiprows_parameter(
         Manage read_csv `skiprows` parameter.
 
         Change `skiprows` parameter in the way Modin could more optimally
-        process it. If `skiprows` is an array, this array will be sorted and
-        then, if array is uniformly distributed, `skiprows` will be "squashed"
-        into integer value and `pre_reading` parameter will be set if needed
-        (in this case fastpath can be done).
+        process it. `csv_dispatcher` have two mechanisms of rows skipping:
+
+        1) During file partitioning (setting of file limits that should be read
+        by each partition) exact rows can be excluded from partitioning scope,
+        thus they won't be read at all and can be considered as skipped. This is
+        the most effective way of rows skipping (since it doesn't require any
+        actual data reading and postprocessing), but in this case `skiprows`
+        parameter can be an integer only. When it possible Modin always uses
+        this approach by setting of `skiprows_partitioning` return value.
+
+        2) Rows for skipping can be dropped after full dataset import. This is
+        more expensive way since it requires extra IO work and postprocessing
+        afterwards, but `skiprows` parameter can be of any non-integer type
+        supported by `pandas.read_csv` `skiprows` parameter. These rows is
+        specified by setting of `skiprows_md` return value.
+
+        In some cases, if `skiprows` is uniformly distributed array (e.g. [1,2,3]),
+        `skiprows` can be "squashed" and represented as integer to make a fastpath.
+        If there is a gap between the first row for skipping and the last line of
+        the header (that will be skipped too), then assign to read this gap first
+        (assign the first partition to read these rows be setting of `pre_reading`
```

Comment:
Can we just pass several pairs (start position, end position) into one parser, and then make the resulting dataframe with `concat` function. This will get rid of situations where partitions are not balanced in terms of the number of rows.

Reply:
This is outside of this pull request, but still I wonder what you think about it.

Reply:
Yes, you are right, we can balance the load between partitions in this way, but it will cause code complication and it is hard to estimate the speed-up we can achieve using this approach (i beware speed up will not be big enough), but anyway let's try. Corresponding issue was created https://github.com/modin-project/modin/issues/3506.